Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -373,89 +373,107 @@ static void bufferWasConsumedCallback(CMNotificationCenterRef, const void* liste
     resetParserState();
 }
 
-void SourceBufferPrivateAVFObjC::didParseInitializationData(InitializationSegment&& segment)
+void SourceBufferPrivateAVFObjC::setTrackChangeCallbacks(const Vector<RefPtr<TrackPrivateBase>>& tracks, bool initialized)
 {
-    ALWAYS_LOG(LOGIDENTIFIER);
+    for (auto& track : tracks) {
+        if (auto videoTrack = dynamicDowncast<VideoTrackPrivate>(track)) {
+            videoTrack->setSelectedChangedCallback([weakThis = WeakPtr { *this }, this, initialized] (VideoTrackPrivate& track, bool selected) {
+                if (!weakThis)
+                    return;
 
-    if (!m_mediaSource) {
-        return;
-    }
+                auto videoTrackSelectedChanged = [weakThis, this, trackRef = Ref { track }, selected] {
+                    if (!weakThis)
+                        return;
+                    trackDidChangeSelected(trackRef, selected);
```

Comment:
Given the lambda is one line, in the initialised case, I would tend to just call this method directly.